launch_testing_ros/examples/talker_listener.test.py

```diff
@@ -146,17 +146,22 @@ def test_listener_receives(self, listener):
         )
         self.addCleanup(self.node.destroy_publisher, pub)
 
-        # Publish some unique messages on /chatter and verify that the listener gets them
-        # and prints them
-        for _ in range(5):
-            msg = std_msgs.msg.String()
-            msg.data = str(uuid.uuid4())
+        # Publish a unique message on /chatter and verify that the listener
+        # gets it and prints it
+        msg = std_msgs.msg.String()
+        msg.data = str(uuid.uuid4())
+        for _ in range(10):
 
             pub.publish(msg)
-            self.proc_output.assertWaitFor(
+            success = self.proc_output.waitFor(
                 expected_output=msg.data,
-                process=listener
+                process=listener,
+                timeout=1.0,
             )
+            if success:
+                break
+        else:
```

Comment:
nit: the loop-else looks odd and might confuse future readers. Maybe just `assert success` instead?

Reply:
Good point: done in d1724b73db122f0ebdf6dcce198c04c528e0bab2.